samples/boards/nrf/s2ram/src/main.c

```diff
@@ -15,24 +15,34 @@
 #include <hal/nrf_regulators.h>
 #include <hal/nrf_gpio.h>
 
-static void system_off(void)
+static int system_off(void)
 {
 	nrf_regulators_system_off(NRF_REGULATORS);
+
+	/*
+	 * We should never reach this point if the system is powered off. If we
+	 * do, return -EBUSY.
+	 */
+	return -EBUSY;
```

Comment:
shouldn't we add here `CODE_UNREACHABLE;` then?

Reply:
well, no, if we think that `nrf_regulators_system_off()` can actually fail to shut the system down. In this case we want to return some negative value (`-EBUSY` or anything else really) to signal this case to the caller.

Reply:
can it fail without returning any error? :/

Reply:
uhm nope because This value is actually returned in chain by `arch_pm_s2ram_suspend()` so that we can differentiate between: (1) correctly resuming from suspend vs (2) something went wrong during suspension.

https://github.com/zephyrproject-rtos/zephyr/blob/86c06f626cff41616c0b28178ff5b49e79421ac3/samples/boards/nrf/s2ram/src/main.c#L37-L45